File: Patient/WebContent/Componet/patient.js

$(document).ready(function() {
	$("#alertSuccess").hide();
	$("#alertError").hide();
});

// SAVE ============================================
$(document).on("click", "#btnSave", function(event) {

	// Clear alerts---------------------
	$("#alertSuccess").text("");
	$("#alertSuccess").hide();
	$("#alertError").text("");
	$("#alertError").hide();

	// Form validation-------------------
	var status = validatePatientForm();
	if (status != true) {
		$("#alertError").text(status);
		$("#alertError").show();
		return;
	}

	var type = ($("#hidPatientIDSave").val() == "") ? "POST" : "PUT";

	$.ajax({
		url : "PatientAPI",
		type : type,
		data : $("#paticientform").serialize(),
		dataType : "text",
		complete : function(response, status) {
			console.log(response)
			onPatientSaveComplete(response.responseText, status);
		}
	});
});

function onPatientSaveComplete(response, status) {
	if (status == "success") {
		var resultSet = JSON.parse(response);
		if (resultSet.status.trim() == "success") {
			$("#alertSuccess").text("Successfully saved.");
			$("#alertSuccess").show();
			$("#divPatientGrid").html(resultSet.data);
		} else if (resultSet.status.trim() == "error") {
			$("#alertError").text(resultSet.data);
			$("#alertError").show();
		}
	} else if (status == "error") {
		$("#alertError").text("Error while saving.");
		$("#alertError").show();
	} else {
		$("#alertError").text("Unknown error while saving..");
		$("#alertError").show();
	}
	$("#hidPatientIDSave").val("");
	$("#P_Id")[0].reset();
}

//UPDATE==========================================
$(document).on("click", ".btnUpdate", function(event)
		{
			$("#hidPatientIDSave").val($(this).closest("tr").find('td:eq(0)').text());
			$("#fNmae").val($(this).closest("tr").find('td:eq(1)').text());
			$("#lName").val($(this).closest("tr").find('td:eq(2)').text());
			$("#gender").val($(this).closest("tr").find('td:eq(3)').text());
			$("#age").val($(this).closest("tr").find('td:eq(4)').text());
			$("#patientNIC").val($(this).closest("tr").find('td:eq(5)').text());
			$("#address").val($(this).closest("tr").find('td:eq(6)').text());
			$("#patientEmail").val($(this).closest("tr").find('td:eq(7)').text());
			$("#passwod").val($(this).closest("tr").find('td:eq(8)').text());
			$("#phoneNumber").val($(this).closest("tr").find('td:eq(9)').text());

		});


//Remove
$(document).on("click", ".btnRemove", function(event)
		{
		 	$.ajax(
		{
			url : "PatientAPI",
			type : "DELETE",
			data : "P_Id=" + $(this).val(),
			dataType : "text",
			complete : function(response, status)
			{
				onPatientDeleteComplete(response.responseText, status);
			}
		 });
		});


	function onPatientDeleteComplete(response, status)
	{
		if (status == "success")
	{
			var resultSet = JSON.parse(response);
			if (resultSet.status.trim() == "success")
	{
			$("#alertSuccess").text("Successfully deleted.");
			$("#alertSuccess").show();
			$("#divPatientGrid").html(resultSet.data);
	} 
	else if (resultSet.status.trim() == "error")
	{
			$("#alertError").text(resultSet.data);
			$("#alertError").show();
	}
	}
	else if (status == "error")
	{
		$("#alertError").text("Error while deleting.");
		$("#alertError").show();
	} 
	else
	{
		$("#alertError").text("Unknown error while deleting..");
		$("#alertError").show();
	}
}
// CLIENT-MODEL================================================================
function validatePatientForm() {
	// First name
	if ($("#fNmae").val().trim() == "") {
		return "Insert First Name";
	}

	// Last name
	if ($("#lName").val().trim() == "") {
		return "Insert Last Name.";
	}
	// Gender
	if ($("#gender").val().trim() == "") {
		return "Insert Gender.";
	}
	// Age
	if ($("#age").val().trim() == "") {
		return "Insert Age.";
	}

	// NIC
	if ($("#patientNIC").val().trim() == "") {
		return "Insert NIC.";
	}
	// address
	if ($("#address").val().trim() == "") {
		return "Insert address.";
	}
	// email
	if ($("#patientEmail").val().trim() == "") {
		return "Insert Email.";
	}
	// password
	if ($("#passwod").val().trim() == "") {
		return "Insert Password.";
	}
	// phone number
	if ($("#phoneNumber").val().trim() == "") {
		return "Insert Phone number.";
	}
	return true;
}

File: Patient/src/model/Patient.java
package model;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.Statement;


public class Patient {

	// A common method to connect to the DB
	private Connection connect() {
		Connection con = null;

		try {
			Class.forName("com.mysql.jdbc.Driver");

			// Provide the correct details: DBServer/DBName, username, password
			con = DriverManager.getConnection("jdbc:mysql://127.0.0.1:3306/helthcare?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC", "root", "");
			System.out.println("Connected Database");
		} catch (Exception e) {
			e.printStackTrace();
		}

		return con;
	}

	//---------------------------Insert--------------------

	public String AddPatient(String P_fname, String P_lname, String P_gender, String P_age, String P_patientNIC,
			String P_address, String P_email, String P_password , String P_phoneNo) {
		String output = "";

		try {
			Connection con = connect();

			if (con == null) {
				return "Error while connecting to the database for inserting.";
			}

			// create a prepared statement
			String query = " INSERT INTO `patient`(`fNmae`, `lName`, `gender`, `age`, `patientNIC`, `address`, `patientEmail`, `passwod`,`phoneNumber`) "
					+ "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)";

			PreparedStatement preparedStmt = con.prepareStatement(query);

			
			// binding values
			preparedStmt.setString(1, P_fname);
			preparedStmt.setString(2, P_lname);
			preparedStmt.setString(3, P_gender);
			preparedStmt.setString(4, P_age);
			preparedStmt.setString(5, P_patientNIC);
			preparedStmt.setString(6, P_address);
			preparedStmt.setString(7, P_email);
			preparedStmt.setString(8, P_password);
			preparedStmt.setString(9, P_phoneNo);

			// execute the statement
			preparedStmt.execute();
			con.close();
			
			String newpatient = readPatient();
			output = "{\"status\":\"success\"}"; 
			 
		} catch (Exception e) {
			output = "{\"status\":\"error\", \"data\": "
					+ "\"Error while inserting the Patient.\"}"; 
			output = "Error while inserting the Patient Details.";	
			System.err.println(e.getMessage());
		}
		return output;
	}

	// ------------------------------Read----------------------------------

	public String readPatient()  
	{   String output = ""; 
	 
	  try   { 
		  Connection con = connect(); 
		  
		   if (con == null)    
		   {return "Error while connecting to the database for reading."; } 
		 
		   // Prepare the html table to be displayed    
		   output = "<table border=\"1\"><tr><th>Patient ID</th><th>Patient Fname</th><th>Patient Lname</th><th>"
		   		+ "Patient Gender</th><th>Patient Age</th><th>Patient NIC</th><th>Patient Address"
		   		+ "</th><th>Patient Email</th><th>Patient Password</th><th>Patient PhoneNo</th><th>Update</th>"
		   		+ "<th>Remove</th></tr>"; 
		 
		   String query = "select * from patient WHERE `status` = 'active';"; 
		   
		   Statement stmt = con.createStatement();    
		   ResultSet rs = stmt.executeQuery(query); 
		   
		   // iterate through the rows in the result set    
		   while (rs.next())    
		   {   
			   String P_Id = Integer.toString(rs.getInt("patientId"));     
			   String P_fname = rs.getString("fNmae");     
			   String P_lname = rs.getString("lName");     
			   String P_gender = rs.getString("gender");     
			   String P_age = Integer.toString(rs.getInt("age"));     
			   String P_patientNIC = rs.getString("patientNIC"); 
			   String P_address = rs.getString("address");
			   String P_email = rs.getString("patientEmail");
			   String P_password = rs.getString("passwod");
			   String P_phoneNo = Integer.toString(rs.getInt("phoneNumber"));
		   
		  
		   
		    // Add into the html table
			   	output += "<tr><td><input id='hidPatientIdUpdate'name='hidPatientIdUpdate' type='hidden' value='"+P_Id+"'>" + P_Id + "</td>";
		   		output += "<td>" + P_fname + "</td>";     
		   		output += "<td>" + P_lname + "</td>";    
		   		output += "<td>" + P_gender + "</td>";     
		   		output += "<td>" + P_age + "</td>"; 
		   		output += "<td>" + P_patientNIC + "</td>";
		   		output += "<td>" + P_address + "</td>";
		   		output += "<td>" + P_email + "</td>";
		   		output += "<td>" + P_password + "</td>";
		   		output += "<td>" + P_phoneNo + "</td>";
		 
		    // buttons     
		   		/*output += "<td><input name=\"btnUpdate\" type=\"button\" "
		   				+ "value=\"Update\" class=\"btn btn-secondary\"></td>"      
		   				+ "<td><form method=\"post\" action=\"Patient.jsp\">"      
		   				+ "<input name=\"btnRemove\" type=\"submit\" value=\"Remove\"     "
		   				+ " class=\"btn btn-danger\">"      
		   				+ "<input name=\"hidpatientIdDelet\" type=\"hidden\" value=\"" + P_Id      
		   				+ "\">" + "</form></td></tr>";*/ 
		   		output += "<td><input name='btnUpdate' type='button' value='Update' class='btnUpdate btn btn-secondary'></td> "
						+ "<td><button name='btnRemove' type='button' value='"+ P_Id + "' class='btnRemove btn btn-danger'  data-ID= '" + P_Id + "'>remove</button></td></tr>";
		   }
	  
	 // Connection con;
	con.close();
		// Complete the html table
		output += "</table>";
	} catch (Exception e) {
		output = "Error while reading the items.";
		System.err.println(e.getMessage());
	}
	return output;
}

	//--------------------------Update ---------------------

	public String updatePatient(String P_Id, String P_fname, String P_lname, String P_gender, String P_age, String P_patientNIC, String P_address, String P_email, String P_password, String P_phoneNo ) {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for updating.";
			}
			// create a prepared statement
			
			String query = "UPDATE patient SET  fNmae =?, lName=? ,gender=?, age=?, patientNIC=?, address=?, patientEmail=?, passwod=?, phoneNumber=? "
					+ "WHERE patientId=? AND  `status` = 'active'";
			
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			preparedStmt.setString(1, P_fname);
			preparedStmt.setString(2, P_lname);
			preparedStmt.setString(3, P_gender);
			preparedStmt.setInt(4, Integer.parseInt(P_age));
			preparedStmt.setString(5, P_patientNIC);
			preparedStmt.setString(6, P_address);
			preparedStmt.setString(7, P_email);
			preparedStmt.setString(8, P_password);
			preparedStmt.setInt(9, Integer.parseInt(P_phoneNo));
			preparedStmt.setInt(10, Integer.parseInt(P_Id));


			
			// execute the statement
			preparedStmt.execute();
			con.close();
			String newDoctor = readPatient();
			output = "{\"status\":\"success\"}"; 
			 
		} catch (Exception e) {
			output = "Error while updating the item.";
			System.err.println(e.getMessage());
		}
		return output;
	}

	// -------------------------------Delete-------------------------

	public String deletePatient(String patientId) {
		String output = "";

		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for deleting.";
			}

			
			String query = "UPDATE patient SET status = 'Deleted' "
					+ "WHERE patientId=?";
			
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			
			preparedStmt.setInt(1, Integer.parseInt(patientId));


			// execute the statement
						preparedStmt.execute();
						con.close();
						
						String newDoctor = readPatient();
						output = "{\"status\":\"success\"}";
						 
					} catch (Exception e) {
						output = "{\"status\":\"error\", \"data\": "
								+ "\"Error while deleting the Patient.\"}"; 
						output = "Error while deleting the Patient Details.";	
						System.err.println(e.getMessage());
					}
					return output;
	}
}
